internal/scorecard/helpers/test_definitions.go

```diff
@@ -102,6 +103,17 @@ func (ts *TestSuite) TotalScore() (score int) {
 	return int(floatScore * (100 / addedWeights))
 }
 
+// Apply label selectors removing tests that do not match
+func (ts *TestSuite) ApplySelector(selector labels.Selector) {
+	for i := 0; i < len(ts.Tests); i++ {
+		if !selector.Matches(labels.Set(ts.Tests[i].GetLabels())) {
+			// Remove the test
```

Comment:
We should also delete from `ts.Weights` as well, to reverse the effect of [`AddTests()`](https://github.com/operator-framework/operator-sdk/blob/041ec5463883d336057c84831ddb200b32e73c25/internal/scorecard/helpers/test_definitions.go#L79-L83):
```go
delete(ts.Weights, ts.Tests[i].GetName())
```

And since we would need to use `ts.Tests[i]` to get the labels and name, maybe it's worth it to declare `t := ts.Tests[i]`?

Reply:
added.